pkg/controller/composite/hooks_test.go

```diff
@@ -88,3 +89,27 @@ func TestSyncHookRequest_MarshalJSON(t *testing.T) {
 		t.Errorf("the expected result is not equal to actual: %s", diff)
 	}
 }
+
+func TestWhenChildrenArrayIsNullThenDeserializeToEmptySlice(t *testing.T) {
+	input := `
+{
+	"children": [null]
+}`
+	parentController := parentController{
+		syncHook: hooks.NewSerializingExecutorStub(input),
+	}
+	parent := &unstructured.Unstructured{}
+	parent.SetDeletionTimestamp(nil)
+	request := SyncHookRequest{Parent: parent}
+
+	response, err := parentController.callHook(&request)
+
+	if err != nil {
+		t.Error(err)
+		t.Fail()
+	}
+
+	if response.Children == nil {
+		t.Errorf("Children should not be nil")
```

Comment:
I guess the response can be set by the handler code. In other words the children field can be set to null outside of metacontroller.

Am I right in above thinking. Can we just implement a nil check to avoid the original issue of crash.

Reply:
hi, unsure if I am following, to avoid `nil` I initialize response with `empty` Children/Attachements` (i.e. `response := SyncHookResponse{Children: []*unstructured.Unstructured{}}`) so that field cannot be `nil`, thus additional check is not needed. I choosed to use this way instead of introducing `if ...` statement as it seems simpler.

Reply:
Well, I believe the `if` condition is still required to handle cases where hook implementation itself sets the response.children to nil.

For example, one can always set the children to nil here https://github.com/metacontroller/metacontroller/blob/master/examples/daemonjob/manifest/sync.py#L71-L75

Reply:
actually this PR is fixing that behavior, if the user controller sets `nil` there, with the current initialization it will be defaulted to empty array. The test I wrote is testing that case.

Reply:
Ok.. Great then.